processing/src/main/java/io/druid/query/filter/DimFilterUtils.java

```diff
@@ -122,16 +122,19 @@
       boolean include = true;
 
       if (dimFilter != null && shard != null) {
-        Map<String, RangeSet<String>> domain = shard.getDomain();
-        for (Map.Entry<String, RangeSet<String>> entry : domain.entrySet()) {
-          String dimension = entry.getKey();
+        Map<String, RangeSet<String>> filterDomain = Maps.newHashMap();
```

Comment:
nit: there's no point to use the Guava API. Please use `new HashMap<>()` instead.